x/mongo/driver/operation.go

```diff
@@ -446,7 +446,19 @@ func (op Operation) Execute(ctx context.Context, scratch []byte) error {
 			if e := err.(Error); retryable && op.Type == Write && e.UnsupportedStorageEngine() {
 				return ErrUnsupportedStorageEngine
 			}
-			if retryable && tt.Retryable() && retries != 0 {
+
+			connDesc := conn.Description()
+			retryableErr := tt.Retryable()
+			if op.Type == Write {
+				retryableErr = tt.RetryableWrite(connDesc.WireVersion)
+			}
+
+			// Add a RetryableWriteError label for retryable errors from pre-4.4 servers
+			if tt.HasErrorLabel(NetworkError) || (retryableErr && connDesc.WireVersion != nil && connDesc.WireVersion.Max < 9) {
```

Comment:
This label should only be added if `op.Type == Write`. Also I don't think you need to check for `NetworkError` here because the `tt.RetryableWrite` call already checks for it.

So I think this logic should just be `if wireVersion < 9 && op.Type == Write && retryableErr ...`

Reply:
Also, I can't comment this in the `WriteCommandError` case because that code wasn't changed, but the spec says that this label should also be added for "a server response with a write concern error response containing any of the previously listed codes". Do we need a similar conditional in the case above?